pkg/vault/test/authorizer/AuthorizerWithAdaptorValidation.test.ts

```diff
@@ -86,14 +87,37 @@ describe('AuthorizerWithAdaptorValidation', () => {
         expect(await authorizer.connect(other).canPerform(ROLE_1, other.address, ANY_ADDRESS)).to.be.false;
       });
     });
-  });
 
-  describe('Adaptor and Entrypoint interactions', () => {
-    sharedBeforeEach('upgrade to the new authorizer', async () => {
-      await actualAuthorizer.connect(admin).grantRole(await actionId(vault, 'setAuthorizer'), admin.address);
-      await vault.connect(admin).setAuthorizer(authorizer.address);
-    });
+    describe('Adaptor and Entrypoint interactions (post-upgrade)', () => {
+      sharedBeforeEach('upgrade to the new authorizer', async () => {
+        await actualAuthorizer.connect(admin).grantRole(await actionId(vault, 'setAuthorizer'), admin.address);
+        await vault.connect(admin).setAuthorizer(authorizer.address);
+      });
+
+      it('adaptor calls from entrypoint contract succeed', async () => {
+        expect(await authorizer.connect(adaptorSigner).canPerform(ROLE_1, adaptorEntrypoint.address, ANY_ADDRESS)).to.be
+          .true;
```

Comment:
This is not strictly enough because `canPerform` is now more complex and depends on its caller. We should make actual permissioned calls that go via the adaptor and entrypoint to trigger this behavior.